### PR TITLE
Network Specific Asset Education

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Description**

This PR updates a user when they toggle to a different network and have no token/funds present.
https://app.zenhub.com/workspaces/metamask-mobile-5f984938ddc0e4001d4b79cb/issues/metamask/mobile-planning/149

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

**Issue**

Resolves #

### CODE REVIEW

File: app/components/UI/NetworkList/index.js [hunk 1]
```diff
@@ -136,11 +136,16 @@ export class NetworkList extends PureComponent {
...
+		/**
+		 * A function that handles the network selection
+		 */
+		onNetworkSelected: PropTypes.func,
 	};
 
 	getOtherNetworks = () => getAllNetworks().slice(1);
 
 	onNetworkChange = (type) => {
+		this.props.onNetworkSelected(type, 'ETH');
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/components/UI/NetworkList/index.js]: Updated

File: app/components/UI/NetworkInfo/index.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+	nativeToken: string;
+}
+
+interface DescriptionProps {
+	description: string;
+	clickableText: string | undefined;
+	number: number;
+}
+
+const learnMoreUrl = 'https://metamask.zendesk.com/hc/en-us/articles/4404424659995';
```
Comment (sethkfman) [inline app/components/UI/NetworkInfo/index.tsx:109]: Suggest pulling this from strings in the even we have other translations.

File: app/components/UI/NetworkInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+						</Text>
+					)}
+				</Text>
+			</View>
+			<LineDivide />
+		</View>
+	);
+};
+
+const UnknownAsset = () => (
```
Comment (sethkfman) [inline app/components/UI/NetworkInfo/index.tsx:146]: We typically embed the smaller components inside as render functions that uses the useEffect hook to reduce renders. If you want to keep `UnknownAsset` & `Description` as a separate component move them to another file.

File: app/reducers/networkSelector/index.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+export const NETWORK_ONBOARDED = 'NETWORK_ONBOARDED';
+
+export const initialState = {
+	networkOnboardedState: [],
+};
+
+export const onboardNetworkAction = (data) => ({
```
Comment (sethkfman) [inline app/reducers/networkSelector/index.js:7]: We typically have separate files for the actions. Move this action to the `app/actions`

File: app/util/custom-gas/index.js [hunk 5]
```diff
@@ -11,6 +11,7 @@ const { hexToBN } = util;
 export const ETH = 'ETH';
 export const GWEI = 'GWEI';
 export const WEI = 'WEI';
+export const PRIVATENETWORK = 'PRIVATENETWORK';
```
Comment (sethkfman) [inline app/util/custom-gas/index.js:14]: Consider moving this to the `app/constants/network.js`

**Review by sethkfman — COMMENTED** [review]
A few minor comments around coding style/project arch. Overall looks good.

File: app/util/custom-gas/index.js (hunk 5 above)
Comment (blackdevelopa) [reply app/util/custom-gas/index.js]: Updated. Thank you

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by sethkfman — APPROVED** [review]
LGTM

File: app/components/UI/NetworkInfo/index.tsx (hunk 3 above)
Comment (sethkfman) [reply app/components/UI/NetworkInfo/index.tsx]: Completed

File: app/components/UI/NetworkInfo/index.tsx (hunk 2 above)
Comment (sethkfman) [reply app/components/UI/NetworkInfo/index.tsx]: Completed

**Review by cortisiko — CHANGES_REQUESTED** [review]
Issue 1
when I:
go to chainlist.org
add a network (I used avalanche)
I am taken back to the wallet view and I do not see [the educational modal ](http://recordit.co/xOnL57mjTi)

Issue 2
[After importing a network manually the education modal does not appear.](http://recordit.co/r9uXmiWO7A)

to reproduce:
open the drawer
go to the networks page and tap on "add a network"
Input these into the empty fields: Network name: xDai, RPC url: https://dai.poa.network/, Symbol: xDai, Chain ID: 100 
tap on Add token
notice the network is added, you are on the wallet view but the education modal does not appear.

**Comment by blackdevelopa** [comment]
Closing this PR and updating this #3910

**Pull request closed without merging**